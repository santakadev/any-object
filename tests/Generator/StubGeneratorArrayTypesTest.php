<?php

namespace Santakadev\AnyObject\Tests\Generator;

use ApprovalTests\Approvals;
use Santakadev\AnyObject\Generator\StubGenerator;
use Santakadev\AnyObject\Tests\AnyObjectTestCase;
use Santakadev\AnyObject\Tests\TestData\ArrayTypes\GenericArrayOfStringObject;

class StubGeneratorArrayTypesTest extends AnyObjectTestCase
{
    public function test_generator_generic_array_of_string(): void
    {
        $this->markTestIncomplete();
        $generator = new StubGenerator();
        $text = $generator->generate(GenericArrayOfStringObject::class);
        Approvals::verifyString($text);
        $object = AnyGenericArrayOfStringObject::build();
        $this->assertIsArray($object->value);
        $this->assertGreaterThanOrEqual(0, count($object->value));
        $this->assertLessThanOrEqual(50, count($object->value));
        foreach ($object->value as $item) {
            $this->assertIsString($item);
        }
    }
}
