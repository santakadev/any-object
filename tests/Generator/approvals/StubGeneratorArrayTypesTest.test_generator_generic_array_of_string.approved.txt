<?php

namespace Santakadev\AnyObject\Tests\Generator\Generated;

use Faker\Factory;
use Santakadev\AnyObject\Tests\TestData\ArrayTypes\GenericArrayOfStringObject;

final class AnyGenericArrayOfStringObject
{
    public static function with(array|ValueNotProvided $value = new ValueNotProvided()) : GenericArrayOfStringObject
    {
        if ($value instanceof ValueNotProvided) {
            $faker = Factory::create();
            $minElements = 0;
            $maxElements = 50;
            $elementsCount = $faker->numberBetween($minElements, $maxElements);
            $value = [];
            for ($i = 0; $i < $elementsCount; $i++) {
                $value[] = $faker->text();
            }
        }
        return new GenericArrayOfStringObject($value);
    }
    public static function build() : GenericArrayOfStringObject
    {
        return self::with();
    }
}
